dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1449,7 +1512,8 @@ winrt::TeachingTipPlacementMode TeachingTip::DetermineEffectivePlacement()
             }
         }
     }
-    // The teaching tip wont fit anywhere, just return top.
+    // The teaching tip wont fit anywhere, set tipDoesNotFit and return top to indicate that we should not open.
+    m_tipDoesNotFit = true;
     return winrt::TeachingTipPlacementMode::Top;
```

Comment:
consider returning std::tuple<TeachingTipPlacementMode, bool> -- C++17 added [nice syntax for accepting tuple return values](https://en.cppreference.com/w/cpp/language/structured_binding).